### DISCUSSION TITLE
filter a test on serializable TheoryData with dotnet test

### DISCUSSION THREAD

**Discussion by AlexandreBossard** [xunit/xunit#2938] [Question]
I have the following tests:
```
OperationsControllerTests.Authenticates(data: OperationRequestData { Url = balance, Method = GET, Body =  })
OperationsControllerTests.Authenticates(data: OperationRequestData { Url = nonce, Method = GET, Body =  })
OperationsControllerTests.Authenticates(data: OperationRequestData { Url = list, Method = POST, Body = ListRequest { Limit = 2 } })
```

I would like to filter one of these with `dotnet test --filter` but without success right now. The obvious `dotnet test --filter 'OperationsControllerTests.Authenticates(data: OperationRequestData { Url = balance, Method = GET, Body =  })'` does not work. Is this suppose to work ? what amount of escaping I need ?

I can launch a particular test through vs-code test explorer, so I suppose there is a way to do that.

**Comment by bradwilson** [comment]
Per [the docs](https://learn.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests?pivots=mstest#syntax), the filter you're looking for is `"DisplayName=..."`. Note that I had to escape the parenthesis with backslashes or I'd get a message that said `Incorrect format for TestCaseFilter`:

![image](https://github.com/xunit/xunit/assets/16944/e55d9287-23cf-4fc1-9f24-84f6c9a76931)

**Reply by AlexandreBossard** [reply]
This works with simple values, but as you can see in my example, I use an object for my test parameter. Escaping the braces does not help.

**Reply by bradwilson** [reply]
This is something you're going to have to take up with the VSTest team. They're the ones who have defined this filtering syntax, so they'd be the ones to tell you what it would take to be able to properly filter the test.

https://github.com/microsoft/vstest/issues
